src/command/publish.rs

```diff
@@ -1,15 +1,47 @@
+use self::Access::{Public, Restricted};
 use command::utils::{find_pkg_directory, set_crate_path};
 use error::Error;
 use npm;
 use slog::Logger;
 use std::path::PathBuf;
 use std::result;
+use std::str::FromStr;
 use PBAR;
 
+#[derive(Debug)]
+pub enum Access {
+    Public,
+    Restricted,
+}
+
+impl FromStr for Access {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        match s {
+      "public" => Ok(Access::Public),
+      "restricted" => Ok(Access::Restricted),
+      "private" => Ok(Access::Restricted),
+      _ => Err(Error::Unsupported { message: format!("{} is not a supported access level. See https://docs.npmjs.com/cli/access for more information on npm package access levels.", s)}),
+    }
+    }
+}
+
 /// Creates a tarball from a 'pkg' directory
 /// and publishes it to the NPM registry
-pub fn publish(path: Option<PathBuf>, log: &Logger) -> result::Result<(), Error> {
+pub fn publish(
+    path: Option<PathBuf>,
+    access: Option<Access>,
+    log: &Logger,
+) -> result::Result<(), Error> {
     let crate_path = set_crate_path(path);
+    let access = match access {
+        Some(access) => match access {
+            Public => "public",
+            Restricted => "restricted",
+        },
+        None => "",
+    };
```

Comment:
How about instead of making this stringly typed again, we implement `Display` for `Access`, implement `Default` for `Access`, and then right here do

```rust
access.unwrap_or_default()
```

Reply:
I agree with the use of `Display` to not have to worry about strings and have it be centralized to one spot so that calls to `to_string()` will be consistent as well

Reply:
so this actually isn't stringly typed- or it is, only insofar as struct opt needs to read strings from the CLI. once it's read from the CLI, it's treated as the enum, and if anything is passed that doesn't work, it's rejected. we can be sure that we never pass npm publish a bad string. that being said i can refactor this a bit, but just wanted to be clear it's not stringly typed. additionally, assuming we don't want to override npm defaults, we can't actually set a default for access because it will be based on whether the package is scoped which we don't track (and is very difficult to track since the package.json can be edited at anytime between init/build and publish)...

Reply:
(i do like the idea of implement Display for it tho, i just don't think that removes having to implement the from string because of struct opt)

Reply:
the one thing to note is that we can't implement a default- this is because the default changes depending on whether or not the package is scoped, which we don't currently track. i need to think about whether i think it's a good idea to do so, but i had a plan for how to do the default that didn't involve tracking whether it was scoped so i may try that first.